src/openms/include/OpenMS/FILTERING/NOISEESTIMATION/SignalToNoiseEstimator.h

```diff
@@ -106,46 +109,58 @@ namespace OpenMS
 
 
     /// Set the start and endpoint of the raw data interval, for which signal to noise ratios will be estimated immediately
-    virtual void init(const PeakIterator & it_begin, const PeakIterator & it_end)
-    {
-      first_ = it_begin;
-      last_ = it_end;
-      computeSTN_(first_, last_);
-      is_result_valid_ = true;
-    }
+//    virtual void init(const PeakIterator & it_begin, const PeakIterator & it_end)
+//    {
+//      first_ = it_begin;
+//      last_ = it_end;
+//      computeSTN_(first_, last_);
+//      is_result_valid_ = true;
+//    }
 
     /// Set the start and endpoint of the raw data interval, for which signal to noise ratios will be estimated immediately
     virtual void init(const Container & c)
     {
-      init(c.begin(), c.end());
+      computeSTN_(c);
+      is_result_valid_ = true;
+//init(c.begin(),c.end());
     }
 
     /// Return to signal/noise estimate for data point @p data_point
     /// @note the first query to this function will take longer, as
     ///       all SignalToNoise values are calculated
     /// @note you will get a warning to stderr if more than 20% of the
     ///       noise estimates used sparse windows
-    virtual double getSignalToNoise(const PeakIterator & data_point)
-    {
-      if (!is_result_valid_)
-      {
-        // recompute ...
-        init(first_, last_);
-      }
-
-      return stn_estimates_[*data_point];
-    }
-
-    virtual double getSignalToNoise(const PeakType & data_point)
-    {
-      if (!is_result_valid_)
+      virtual double getSignalToNoise(const Size index)
```

Comment:
Maybe call `const Size index` something that is related to `data_point`. Index doesn't really state anything about the variable's functionality.

Reply:
You can see the last answer of mine.